build.gradle

```diff
@@ -74,25 +74,11 @@ java {
     withSourcesJar()
 }
 
-configurations.configureEach {
-    resolutionStrategy.eachDependency { DependencyResolveDetails details ->
-        // This is a workaround for grails-shell exposing a different version of Groovy
-        // than the one used by Gradle. This causes issues with the Groovy compiler.
-        if (details.requested.group == 'org.apache.groovy') {
-            details.useTarget(group: "org.codehaus.groovy", name: details.requested.name, version: GroovySystem.version)
-            details.because "Use Groovy 3.0.x library shipped with Gradle"
-        }
-
-        if (details.requested.group == 'org.codehaus.groovy') {
-            details.useTarget(group: "org.codehaus.groovy", name: details.requested.name, version: GroovySystem.version)
-            details.because "Use Groovy 3.0.x library shipped with Gradle"
-        }
-    }
-}
+configurations.all*.exclude group: 'org.apache.groovy', module: 'groovy-xml'
```

Comment:
I find It's better to use `configurations.configureEach` as `configurations.all` will configure all configurations eagerly (even unused configurations), while the former does so only lazily upon usage. I also like to write a comment as to why, I for example exclude something, because then it's really easy for the next developer reading the file to understand.